hail/python/test/hailtop/batch/test_batch.py

```diff
@@ -1229,3 +1229,45 @@ def test_update_batch_from_batch_id(self):
         res = b2.run()
         res_status = res.status()
         assert res_status['state'] == 'success', str((res_status, res.debug_info()))
+
+    def test_list_recursive_resource_extraction_in_python_jobs(self):
+        b = self.batch(default_python_image=PYTHON_DILL_IMAGE)
+
+        def write(paths):
+            for path in paths:
+                with open(path, 'w') as f:
+                    f.write('foo')
```

Comment:
nit: can we do `for i, path in enumerate(paths)` and then write `i` to the file so that the files' contents aren't the same?